File: 6. Python/EX39.py
# Função que calcula o fatorial de n
def fatorial(n):

    fat = 1  # Inicializa fatorial

    # Loop para multiplicar de 1 até n
    for i in range(1, n + 1, 1):

        fat *= i  # Multiplica o valor atual por i

    return fat  # Retorna o fatorial calculado

# Função principal
def main():

    while True:  # Loop para entrada de dados

        try:
            n = int(input("# Digite o valor de N: "))  # Lê o valor de N

            # Verifica se n está dentro do limite
            if 0 <= n <= 15:
                print(f"# Fatorial = {fatorial(n)}")  # Exibe o fatorial
                break  # Sai do loop

            else:
                print("# Digite valores abaixo de 15")

        except Exception:
            print("# Erro !")  # Trata entradas inválidas

# Verifica se o script está sendo executado diretamente
if __name__ == '__main__':
    main()  # Chama a função principal

File: 4. Java/EX39.java
package Apostila;

import java.util.Scanner;

public class EX39
{
    public static void main(String[] args)
    {
        int n;

        try (Scanner ent = new Scanner(System.in)) 
        {
            System.out.print("# Digite o valor de N: ");
            n = ent.nextInt();
        }
        
        // Verifica se N é menor ou igual a 15
        if (n <= 15)
        {
            System.out.printf("# Fatorial = %d \n", fatorial(n));            
        }
        
        else
        {
            System.out.println("# Digite valores abaixo de 15 \n");
        }   
    }
    
    // Calcula o fatorial de N
    private static int fatorial(int n)
    {
        int fat = 1;
        
        // Calcula o fatorial usando um loop
        for (int i = 1; i <= n; i++)
        {
            fat *= i;             
        }
        
        return fat;
    }    
}
